### PR TITLE
Task 001

### PR DESCRIPTION
**Pull request by Starlight13**
Refactored code to comform to MVVM + Provider pattern.

### CODE REVIEW

File: lib/screens/shared_components/cart_button.dart [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      }
+    });
+
+    _animationOffset = Tween<Offset>(
+      begin: const Offset(0.0, 0.0),
+      end: const Offset(0.0, -0.5),
+    ).animate(CurvedAnimation(parent: _controller, curve: Curves.bounceIn));
+  }
+
+  @override
```
Comment (ascentman) [inline lib/screens/shared_components/cart_button.dart:37]: try to organise your functions in that way:
- init
- build
- dispose
- private methods, started from underscore

**Review by ascentman — APPROVED** [review]
In general, good work! keep it up!💪

**Comment by Starlight13** [comment]
I have added localisation and refactored the code according to the comments.
I could not find any examples of viewModels communicating with each other in Flutter, and could not figure out how to apply methods used in other programming languages. So I came up with the following solution:
_To remove logic from product_details_screen I added my viewModels to sl, so that I can get them not only from context, but also from there. That way I can avoid passing context to viewModel and remove most of the logic from product_details_screen._ 
Is this solution suitable here?

**Comment by ascentman** [comment]
> I have added localisation and refactored the code according to the comments.
> I could not find any examples of viewModels communicating with each other in Flutter, and could not figure out how to apply methods used in other programming languages. So I came up with the following solution:
> _To remove logic from product_details_screen I added my viewModels to sl, so that I can get them not only from context, but also from there. That way I can avoid passing context to viewModel and remove most of the logic from product_details_screen._
> Is this solution suitable here?

yes, sending viewModels to locator is one of the option and it fits well here. Good job!

File: lib/screens/cart_screen/cart_screen.dart [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        ),
+        actions: [
+          IconButton(
+            onPressed: () {
+              showDialog(
+                context: context,
+                builder: (BuildContext context) {
+                  return AlertDialog(
+                    title: Text(
+                      AppLocalizations.of(context)!.clearCartTitle,
```
Comment (ascentman) [inline lib/screens/cart_screen/cart_screen.dart:37]: you can write something like that: `final localizations = CommonLocalizations.of(context);` below line 19:
`final viewModel = context.watch<ICartViewModel>();` and then use variable `localisations` in all cases. Make this change in all files

**Pull request merged**